### PR TITLE
Fix activity constructor CU-emx4r3

### LINKED ISSUE #8: Activity constructor seems to skip declared measures/ranges
**Issue**
An activity consists of a ```name```, a ```measure``` to express activity metrics, an array of accepted ```ranges```, and the ```oracle``` address. Activities can be enabled by setting it to ```allowed```

But when printing the activity currently deployed it is missing the ```measure``` and ```ranges```.

    // expect(_activity['measures'][0]).to.equal('km');
    // expect(_activity['ranges']).to.equal([2,1024]);

```
[
  'biking',
  '0xA193E42526F1FEA8C99AF609dcEabf30C1c29fAA',
  true,
  name: 'biking',
  oracle: '0xA193E42526F1FEA8C99AF609dcEabf30C1c29fAA',
  allowed: true
]
```

Expected outcome:
1) An activity object has the measure and range
2) Deploy test succeeds with following lines uncommented:
```
    // expect(_activity['measures'][0]).to.equal('km');
    // expect(_activity['ranges']).to.equal([2,1024]);
```

### PR DESCRIPTION
**Pull request by bitbeckers**
* Removed ranges and measures and updated tests
* Updated readme

Fixes #8

### CODE REVIEW

File: contracts/SinglePlayerCommit.sol [hunk 1]
```diff
@@ -271,58 +203,46 @@ contract SinglePlayerCommit is Ownable {
...
-        require(measuresLength == _ranges.length, "SPC::_addActivity - measures and ranges must have same length");
-
-        Activity memory activity;
-
-        activity.name = _name;
-        activity.oracle = _oracle;
-        activity.measures = _measures;
-        activity.ranges = _ranges;
+    function _addActivity(string memory _activityName, address _oracleAddress) internal returns (bytes32 activityKey) {
+        bytes memory activityNameBytes = bytes(_activityName);
```
Comment (spengrah) [inline contracts/SinglePlayerCommit.sol:227]: is this just to check whether the name is empty? could we use line 230 to do the same check? i.e. if `_activityKey has length > 0`?

File: test/SinglePlayerCommit.deploy.ts [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
-import { BytesLike } from "ethers/lib/utils";
+import { BytesLike, solidityKeccak256 } from "ethers/lib/utils";
 
 export function shouldDeployWithInitialParameters(): void {
 
-  it("has the 'biking' activity and it is allowed", async function () {
-    const activityKey: BytesLike = await this.singlePlayerCommit.activityList(0);
-    const _activityName: string = await this.singlePlayerCommit.getActivityName(activityKey);
- 
+  it("has the 'biking' and 'cycling' activity and it is allowed", async function () {
```
Comment (spengrah) [inline test/SinglePlayerCommit.deploy.ts]: 'biking' and 'running'?

**Review by spengrah — APPROVED** [review]

File: contracts/SinglePlayerCommit.sol (hunk 1 above)
Comment (bitbeckers) [reply contracts/SinglePlayerCommit.sol:227]: Just tried it out, but it does not seem to work. I expect this has to do with the empty string also being encoded and hashed using keccak256, resulting in a non-empty variable. When adding "" as one of the activities, deployment succeeds (tried in tests)

File: test/SinglePlayerCommit.deploy.ts (hunk 2 above)
Comment (bitbeckers) [reply test/SinglePlayerCommit.deploy.ts]: fixed

File: contracts/SinglePlayerCommit.sol (hunk 1 above)
Comment (spengrah) [reply contracts/SinglePlayerCommit.sol:227]: ok i see - makes sense

**Pull request merged**